File: chefmate-backend/src/main/java/com/chefmate/backend/entity/ShoppingListItem.java
package com.chefmate.backend.entity;

import jakarta.persistence.*;
import com.chefmate.backend.entity.Recipe;

@Entity
@Table(name = "shopping_list_items")
public class ShoppingListItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String quantity;
    private String unit;
    private Boolean purchased = false;

    @ManyToOne
    @JoinColumn(name = "shopping_list_id")
    private ShoppingList shoppingList;

    @ManyToOne
    @JoinColumn(name = "recipe_id")
    private Recipe recipe;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getQuantity() { return quantity; }
    public void setQuantity(String quantity) { this.quantity = quantity; }

    public String getUnit() { return unit; }
    public void setUnit(String unit) { this.unit = unit; }

    public Boolean getPurchased() { return purchased; }
    public void setPurchased(Boolean purchased) { this.purchased = purchased; }

    public ShoppingList getShoppingList() { return shoppingList; }
    public void setShoppingList(ShoppingList shoppingList) { this.shoppingList = shoppingList; }

    public Recipe getRecipe() { return recipe; }
    public void setRecipe(Recipe recipe) { this.recipe = recipe; }
}

File: ChefMate-Android/app/src/main/java/com/chefmate/data/api/models/ShoppingListItem.kt
package com.chefmate.data.api.models

import com.google.gson.annotations.SerializedName

data class ShoppingListItem(
    @SerializedName("id")
    val id: Long? = null,

    @SerializedName("name")
    val name: String,

    @SerializedName("quantity")
    val quantity: String? = null,

    @SerializedName("unit")
    val unit: String? = null,

    @SerializedName("purchased")
    val purchased: Boolean = false,

    @SerializedName("recipeId")
    val recipeId: Long? = null
) {
    // Backward compatibility - use purchased as checked
    val checked: Boolean
        get() = purchased
}